src/dowhen/trigger.py

```diff
@@ -71,34 +70,62 @@ def _get_code_from_entity(
             if inspect.isfunction(entity) or inspect.ismethod(entity):
                 entity = inspect.unwrap(entity)
                 if inspect.isfunction(entity) or inspect.ismethod(entity):
-                    direct_code_objects.append(entity.__code__)
+                    code_objects.append(entity.__code__)
                 else:  # pragma: no cover
                     raise TypeError(
                         f"Expected a function or method, got {type(entity)}"
                     )
             elif inspect.iscode(entity):
-                direct_code_objects.append(entity)
+                code_objects.append(entity)
             else:
                 raise TypeError(f"Unknown entity type: {type(entity)}")
 
-        for code in direct_code_objects:
-            stack = [code]
-            while stack:
-                current_code = stack.pop()
-                assert isinstance(current_code, CodeType)
+        return code_objects
 
-                all_code_objects.append(current_code)
-                for const in current_code.co_consts:
-                    if isinstance(const, CodeType):
-                        stack.append(const)
+    @classmethod
+    def unify_identifiers(
+        cls,
+        entity: CodeType | FunctionType | MethodType | ModuleType | type | None,
+        *identifiers: IdentifierType | tuple[IdentifierType, ...],
+    ) -> tuple[IdentifierType | tuple[IdentifierType, ...], ...]:
+        """Unify identifiers by resolving relative line numbers."""
+
+        def unify_identifier(
+            entity: CodeType | FunctionType | MethodType | ModuleType | type | None,
+            identifier: IdentifierType,
+        ) -> IdentifierType:
+            if (
+                isinstance(identifier, str)
+                and identifier.startswith("+")
+                and identifier[1:].isdigit()
+            ):
+                if entity is None:
+                    raise ValueError(
```

Comment:
Suppose wo should use a TypeError here.



```suggestion
                    raise TypeError(
```

Reply:
I see～